### ISSUE TITLE
Case with a large lambda

### ISSUE THREAD

**Issue by hyu-kim** [hyu-kim/mds-hypothesis-testing#14]
Refer to a reply / comment to PR #13 

### Issue
Configuration diverges when code runs under 𝝀=10 using Site 1, 2

### Idea
Our confirmatory term in objective function is not regularized and is proportional to hyperparameter. This may have caused a problem with a large hyperparameter.

I will come back to this after completing with outline document.

**Comment by hyu-kim** [comment]
하이퍼파라미터를 Witten의 SMDS가 쓰는 식으로 (alpha, 1-alpha) 아이디어를 적용해 보았는데 통하지 않았다.

**Comment by hyu-kim** [comment]
tested by updating configuration all at once when every samples are done (comment / uncomment line 122 / 127). 
Didn't work still.
This does not make a configuration converge either, when lambda = 1

**Comment by hyu-kim** [comment]
The confirmatory term is not strictly convex in terms of \mathbf{z}, because the coefficient of every quadratic term z_{is}^2 is not always positive. Possibly a reason why MM algorithm does not work at large \lambda.

**Comment by hyu-kim** [comment]
- 알고리즘에 관한 궁금증이 있어 논문을 읽었는데, MM보단 stochastic gradient descent (SGD)가 더 수렴 속도가 빠르다는 흥미로운 주장을 발견했다 [1,2]. 
- 과학자들과 미팅이 끝나면, SGD로 목적함수를 최적화하는 코딩을 짜볼 계획. @soob-kim 
- Objective function을 고쳐서 MM이 가능케 하는 것도 고려 중 (백업 플랜) 

[1] Demaine et al. [Multidimensional Scaling: Approximation and Complexity](http://proceedings.mlr.press/v139/demaine21a.html). ICML (2021)
[2] Zheng et al. [Graph Drawing by Stochastic Gradient Descent](https://arxiv.org/abs/1710.04626). Arxiv (2018)

> The confirmatory term is not strictly convex in terms of \mathbf{z}, because the coefficient of every quadratic term z_{is}^2 is not always positive. Possibly a reason why MM algorithm does not work at large \lambda.

**Comment by hyu-kim** [comment]
수빈 @soob-kim ,

일단 sgd 알고리즘으로 코드를 짜긴 했는데 의외로 수렴이 잘 되지 않더라고
그래서 안되는 이유를 생각해 보았어

1. 코드 또는 pdf 수식 유도에 문제가 있다
2. 초기 값을 랜덤이 아닌 pure MDS로 했기 때문에 눈에 보이는 효과가 미미하다
(FYI – [Zheng et al](https://arxiv.org/pdf/1710.04626.pdf) 이 주장한 논문에선 초기값을 랜덤으로 두었음)

그래서 말인데 1에 관련해서 혹시 시간이 괜찮으면 내가 적은 코드 / 수식이 맞는지 교차 확인을 좀 부탁해도 될까?
코드는 branch `iss14` 에 위치해 있고, 파일 이름은 `sgd.R` 이야 ([링크](https://github.com/hyu-kim/mds-hypothesis-testing/blob/iss14/data/sgd.R))

근데 시간이 없으면 괜찮아, 내가 남은 이번주에 다른 실험을 해야해서 물었어
혹시 내용 팔로업이 어려우면 잠깐 미팅을 해도 괜찮고!

**Comment by soob-kim** [comment]
응 학기 중이라 정신이 없어서 지난 번에 보내준 수식 확인을 아직 못했어,
급한 건이면 이번 주 주말 안으로 살펴볼게, 혹시 급한 게 아니면 다음주 주말까지 확인해봐도 괜찮을까?

**Comment by hyu-kim** [comment]
Closing the issue. Will keep with MM and restrict the range of lamdba for now
